src/powershell/Add-ExportScopeCredential.psm1

```diff
@@ -0,0 +1,109 @@
+<#
+.SYNOPSIS
+Creates secrets in Azure KeyVault for the Service Principal.
+
+.PARAMETER HubName
+The name of the FinOps Hub.  
+Assumes HubName = ResourceGroupName.
+
+.PARAMETER ClientId
+The Application ID (Client ID) of the Service Principal to create the secret for.
+
+.PARAMETER TenantId
+The ID of the Azure AD Tenant ID containing the Service Principal.
+
+.PARAMETER Secret
+The Secret (Password) of the service principal.
+
+.EXAMPLE
+$clientId = ConvertTo-SecureString -AsPlainText -String '00000000-0000-0000-0000-000000000000'
+$secret = ConvertTo-SecureString -AsPlainText -String 'SomePassword'
+Add-ExportScopeCredential -HubName FinOps-Hub  -TenantId $tenantid -ClientId $clientId -Secret $secret
+
+Creates secrets in Azure KeyVault for the Service Principal.
+#>
+function Add-ExportScopeCredential {
+    [CmdletBinding()]
+    param (
+        [Parameter()]
+        [string]
+        [ValidateNotNullOrEmpty()]
+        $HubName,
+        [Parameter()]
+        [String]
+        [ValidateNotNullOrEmpty()]
+        $TenantId,
+        [Parameter()]
+        [Securestring]
+        [ValidateNotNullOrEmpty()]
+        $ClientId,
```

Comment:
Should we use managed identity? Not sure if it's supported.

Reply:
Managed identities are supported, but just like service principals managed identities can't cross tenant boundaries so we'd be locked into the AAD tenant we're deployed to.  If we want to support multiple enrollments or different clouds we need to have an ID local to the tenant which we can use.